### PR TITLE
Set `utf-8` encoding while connecting to the oracle DB

### PR DESCRIPTION
**Pull request by sgandhi1311**
This pull request addresses an issue where special characters (e.g., Japanese characters) in Oracle database tables are not retrieved correctly, resulting in data discrepancies.

# Problem Description
When connecting to an Oracle database, the Oracle tap does not set the encoding to `UTF-8` by default. This can cause special characters, such as Japanese characters, to be retrieved as `?` instead of their actual values. This issue can lead to data discrepancies and affect the accuracy of the extracted data.
Also to write those special characters, the function parameter (=**ensure_ascii**) needs to set **False** within the json.dumps function.

# Solution

- To resolve this issue, this pull request sets the `UTF-8` encoding while connecting to the Oracle database. This ensures that special characters are retrieved correctly and accurately.
- To ensure proper handling of special characters when converting data to JSON, we also set the **ensure_ascii=False** parameter within the **json.dumps** function. This parameter prevents the function from escaping non-ASCII characters.

# Changes Made

- Set the UTF-8 encoding while connecting to the Oracle database.

# Manual QA steps
 - Verified that special characters are retrieved correctly and accurately.
 
# Risks
 - No significant risks
 
# Rollback steps
 - revert this branch

### CODE REVIEW

File: tap_oracle/db.py [hunk 1]
```diff
@@ -11,5 +11,9 @@ def make_dsn(config):
 
 def open_connection(config):
     LOGGER.info("dsn: %s", make_dsn(config))
-    conn = cx_Oracle.connect(config["user"], config["password"], make_dsn(config))
+    conn = cx_Oracle.connect(config["user"], 
+                             config["password"], 
+                             make_dsn(config), 
+                             encoding='UTF-8',     # Specify the encoding for the connection
+                             )
     return conn
```
Comment (RushiT0122) [inline tap_oracle/db.py:19]:
```suggestion
    conn = cx_Oracle.connect(config["user"], 
                             config["password"], 
                             make_dsn(config), 
                             encoding='UTF-8')     # Specify the encoding for the connection
    return conn
```

File: tap_oracle/sync_strategies/incremental.py [hunk 2]
```diff
@@ -5,6 +5,7 @@
 from singer.schema import Schema
 import tap_oracle.db as orc_db
 import tap_oracle.sync_strategies.common as common
+import tap_oracle.sync_strategies.helper as helper
```
Comment (RushiT0122) [inline tap_oracle/sync_strategies/incremental.py:8]:
```suggestion
from tap_oracle.sync_strategies import common
from tap_oracle.sync_strategies import helper
```

File: tap_oracle/sync_strategies/log_miner.py [hunk 3]
```diff
@@ -13,6 +13,7 @@
 import pytz
 import time
 import tap_oracle.sync_strategies.common as common
+import tap_oracle.sync_strategies.helper as helper
```
Comment (RushiT0122) [inline tap_oracle/sync_strategies/log_miner.py:16]:
```suggestion
from tap_oracle.sync_strategies import common
from tap_oracle.sync_strategies import helper
```

File: tap_oracle/sync_strategies/helper.py [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+from singer import RecordMessage
+
+
+def format_message(message,ensure_ascii=True):
+    return json.dumps(message.asdict(), use_decimal=True, ensure_ascii=ensure_ascii)
+
+
+def write_message(message, ensure_ascii=True):
+    sys.stdout.write(format_message(message, ensure_ascii=ensure_ascii) + '\n')
+    sys.stdout.flush()
```
Comment (RushiT0122) [inline tap_oracle/sync_strategies/helper.py:12]: Any reason we are defining new methods than overwrting implementation in singer-python?

**Review by RushiT0122 — CHANGES_REQUESTED** [review]
This may not be directly related, but could we remove the pdb import from all tap code in a separate pull request?

File: tap_oracle/sync_strategies/helper.py (hunk 4 above)
Comment (RushiT0122) [reply tap_oracle/sync_strategies/helper.py:12]: https://github.com/singer-io/singer-python/blob/master/singer/messages.py#L225

File: tap_oracle/db.py (hunk 1 above)
Comment (sgandhi1311) [reply tap_oracle/db.py]: Done

**Review by RushiT0122 — APPROVED** [review]

**Comment by RushiT0122** [comment]
Please fix the cci config.

**Review by RushiT0122 — APPROVED** [review]

**Pull request merged**